### PR TITLE
Made charts, sider, and header more responsive

### PR DESCRIPTION
**Pull request by SanjeevLakhwani**

### CODE REVIEW

**Review by gsfk — COMMENTED** [review]
Good start... most of the tablet views look pretty good. 

The phone views are pretty garbled for me though, here's iPhone 13: 
Screenshot from 2024-10-16 16-06-54


The name is clipped, the logo is mis-aligned, the links are on the edge of the header instead of inside it, and the project name ("is gohan broken") starts to bleed into the "about" text.

File: src/js/components/SiteHeader.tsx [hunk 1]
```diff
@@ -65,16 +67,25 @@ const SiteHeader = () => {
...
-          <img
-            src="/public/assets/branding.png"
-            alt="logo"
-            style={{ height: '32px', verticalAlign: 'middle', transform: 'translateY(-3px)' }}
-            onClick={() => navigate(`/${i18n.language}${scopeToUrl(scopeObj)}`)}
-          />
+        <Space size={isSmallScreen ? 'small' : 'middle'}>
+          {isSmallScreen ? (
+            <img
+              src="/public/assets/icon_small.png"
```
Comment (davidlougheed) [inline src/js/components/SiteHeader.tsx:73]: this neglects the branding for specific instances.

File: src/js/constants/deviceBreakpoints.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export const DeviceBreakpoints = {
```
Comment (davidlougheed) [inline src/js/constants/deviceBreakpoints.ts:1]: since this is a constant, i think it should be uppercase snake-case

File: src/js/index.tsx [hunk 3]
```diff
@@ -30,6 +30,7 @@ import 'leaflet/dist/leaflet.css';
 import 'bento-charts/src/styles.css';
 import './i18n';
 import '../styles.css';
+import { ResponsiveProvider } from '@/components/ResponsiveContext';
```
Comment (davidlougheed) [inline src/js/index.tsx:33]: wrong import location - see comments

File: src/js/components/ResponsiveContext.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+const ResponsiveContext = createContext<ResponsiveContextType>(DefaultResponsiveContext);
+
+interface ResponsiveProviderProps {
+  children: ReactNode;
+}
+
+export const ResponsiveProvider = ({ children }: ResponsiveProviderProps) => {
+  const [isMobile, setIsMobile] = useState<boolean>(false);
+  const [isTablet, setIsTablet] = useState<boolean>(false);
```
Comment (davidlougheed) [inline src/js/components/ResponsiveContext.tsx:24]: initial state should be based on the initial window innerWidth, not false

File: src/js/components/ResponsiveContext.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+  useEffect(() => {
+    const handleResize = () => {
+      setIsMobile(window.innerWidth <= DeviceBreakpoints.MOBILE);
+      setIsTablet(window.innerWidth > DeviceBreakpoints.MOBILE && window.innerWidth <= DeviceBreakpoints.TABLET);
+    };
+    handleResize();
```
Comment (davidlougheed) [inline src/js/components/ResponsiveContext.tsx:31]: instead of this, set initial state as above - factor out above set logic into small functions for isMobile and isTablet

File: src/js/components/SiteHeader.tsx [hunk 6]
```diff
@@ -15,6 +15,7 @@ import { DEFAULT_TRANSLATION, LNG_CHANGE, LNGS_FULL_NAMES } from '@/constants/co
 import { CLIENT_NAME, PORTAL_URL, TRANSLATED } from '@/config';
 
 import ScopePickerModal from './Scope/ScopePickerModal';
+import { useSmallScreen } from '@/components/ResponsiveContext';
```
Comment (davidlougheed) [inline src/js/components/SiteHeader.tsx:18]: move import with other @-imports

File: src/js/constants/deviceBreakpoints.ts (hunk 2 above)
Comment (davidlougheed) [reply src/js/constants/deviceBreakpoints.ts:1]: although if bento routes are also named like this we can keep it for consistency

File: src/styles.css [hunk 7]
```diff
@@ -27,6 +27,10 @@ body {
   border-right: 1px solid #f0f0f0;
 }
 
+.ant-layout-header {
+  padding: 0 !important;
```
Comment (davidlougheed) [inline src/styles.css:31]: this breaks some styling as we discussed

**Review by davidlougheed — CHANGES_REQUESTED** [review]

File: src/js/components/Overview/ChartCard.tsx [hunk 8]
```diff
@@ -30,6 +29,7 @@ const ChartCard: React.FC<ChartCardProps> = memo(({ section, chart, onRemoveChar
   const td = useTranslationDefault();
   const containerRef = useRef<HTMLDivElement>(null);
   const width = useElementWidth(containerRef, chart.width);
+  const CARD_STYLE: React.CSSProperties = { height: '415px', borderRadius: '11px', ...BOX_SHADOW };
```
Comment (davidlougheed) [inline src/js/components/Overview/ChartCard.tsx:32]: why was this moved?

File: src/styles.css [hunk 9]
```diff
@@ -27,6 +27,10 @@ body {
   border-right: 1px solid #f0f0f0;
 }
 
+.ant-layout-header {
```
Comment (davidlougheed) [inline src/styles.css:30]: still not a fan of this change - is !important needed (generally poor practice?)
is there any way we can keep ant design layout by default?

File: src/js/components/SiteHeader.tsx [hunk 10]
```diff
@@ -15,6 +15,7 @@ import { DEFAULT_TRANSLATION, LNG_CHANGE, LNGS_FULL_NAMES } from '@/constants/co
 import { CLIENT_NAME, PORTAL_URL, TRANSLATED } from '@/config';
 
 import ScopePickerModal from './Scope/ScopePickerModal';
+import { useSmallScreen } from '@/components/Util/ResponsiveContext';
```
Comment (davidlougheed) [inline src/js/components/SiteHeader.tsx:18]: this is an @-import, move it above the local import

File: src/js/components/Overview/OverviewDisplayData.tsx [hunk 11]
```diff
@@ -1,13 +1,19 @@
...
+import { Space } from 'antd';
 
 import { disableChart } from '@/features/data/data.store';
-import type { ChartDataField } from '@/types/data';
-import ChartCard from './ChartCard';
+
 import { CHART_WIDTH, GRID_GAP } from '@/constants/overviewConstants';
 
+import { useSmallScreen } from '@/components/Util/ResponsiveContext';
+import ChartCard from './ChartCard';
```
Comment (davidlougheed) [inline src/js/components/Overview/OverviewDisplayData.tsx:10]: move this below @-imports and group @-imports properly

**Review by davidlougheed — CHANGES_REQUESTED** [review]

File: src/js/components/Overview/ChartCard.tsx (hunk 8 above)
Comment (SanjeevLakhwani) [reply src/js/components/Overview/ChartCard.tsx:32]: remnant of a removed change. now back in position.

**Review by davidlougheed — APPROVED** [review]
lgtm
